vortex-array/benches/iter.rs

```diff
@@ -58,6 +61,30 @@ fn vortex_iter_flat(c: &mut Criterion) {
     });
 }
 
+fn vortex_binary_add(c: &mut Criterion) {
```

Comment:
as of `11eae3b4`, the results are:
```
vortex_binary_add       time:   [636.66 µs 643.62 µs 649.49 µs]
arrow_binary_add        time:   [206.01 µs 209.79 µs 213.72 µs]
```
so Arrow is faster, but its a much smaller difference than the previous iterators. I tried some approaches that used `BytesMut` and building everything through a buffer but that was much slower (1-2ms), still trying to restructure things so this isn't the final performance number.